pkg/webhook/validation/uservalidation.go

```diff
@@ -1,17 +1,42 @@
 package validation
 
 import (
+	"context"
+
 	authenticationv1 "k8s.io/api/authentication/v1"
+	"k8s.io/klog/v2"
 	"k8s.io/utils/strings/slices"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fleetv1alpha1 "go.goms.io/fleet/apis/v1alpha1"
 )
 
 const (
 	mastersGroup = "system:masters"
 )
 
-// TODO:(Arvindthiru) Get valid usernames as flag and allow those usernames.
-
 // ValidateUserForCRD checks to see if user is authenticated to make a request to modify fleet CRDs.
 func ValidateUserForCRD(userInfo authenticationv1.UserInfo) bool {
 	return slices.Contains(userInfo.Groups, mastersGroup)
 }
+
+// ValidateUserForFleetCR checks to see if user is authenticated to make a request to modify Fleet CRs.
+func ValidateUserForFleetCR(ctx context.Context, client client.Client, whiteListedUsers []string, userInfo authenticationv1.UserInfo) bool {
```

Comment:
is this used anywhere?

Reply:
is the whiteList only useful fro FleetCR?

Reply:
Using whiteList for CRD check as well